src/components/TodoTable.vue

```diff
@@ -1,127 +1,83 @@
 <script setup lang="ts">
-import axios from "axios"
-import { onMounted, ref } from "vue"
+import { computed, onMounted, ref } from "vue";
+import axios from "axios";
+import { RouterLink, useRouter } from "vue-router";
+import { deleteTodo } from '../apicalls'
+
+const router = useRouter();
 
 type TodoData = {
-  isComplete: boolean
-  todoName: string
-  createdAt: string
-  updatedAt: string
-  __v: number
-  _id: string
-}
+  isComplete: boolean;
+  todoName: string;
+  createdAt: string;
+  updatedAt: string;
+  __v: number;
+  _id: string;
+};
 
 type TodoResponse = {
-  code: number
-  data: TodoData[]
-}
-
-type TodoObject = Omit<TodoData, "createdAt" | "updatedAt" | "__v" | "_id">
-
-const title = ref("")
-const complete = ref(false)
-const allData = ref<TodoData[]>([])
-const searchParam = ref("")
-const holderArr = ref<TodoData[]>([])
-const showError = ref(false)
+  code: number;
+  data: TodoData[];
+};
+const isCompleteFilter = ref(true);
+const allData = ref<TodoData[]>([]);
+const searchParam = ref("");
+const showArrType = ref<"search" | "filter">("search");
+const searchArr = computed(() => {
+  return allData.value.filter((x) =>
+    x.todoName.toLowerCase().includes(searchParam.value.toLowerCase())
+  );
+});
+const filteredArr = computed(() => {
+  return allData.value.filter(
+    (task) => task.isComplete === isCompleteFilter.value
+  );
+});
 
 const getTodos = async () => {
   try {
     const { data } = await axios.get<TodoResponse>(
       "https://calm-plum-jaguar-tutu.cyclic.app/todos"
-    )
-    console.log(data)
-    return data.data
-  } catch (error) {
-    throw error
-  }
-}
-
-const addTodo = async (requestBody: TodoObject) => {
-  if (!requestBody.isComplete || !requestBody.todoName) {
-    showError.value = true
-    return
-  }
-  try {
-    await axios.post<TodoData>(
-      "https://calm-plum-jaguar-tutu.cyclic.app/todos",
-      {
-        todoName: requestBody.todoName,
-        isComplete: requestBody.isComplete,
-      }
-    )
-    location.reload()
+    );
+    console.log(data);
+    return data.data;
   } catch (error) {
-    console.error(error)
+    throw error;
   }
-}
+};
 
-const updateTodo = async (id: string, isComplete: boolean) => {
-  try {
-    await axios.put(`https://calm-plum-jaguar-tutu.cyclic.app/todos/${id}`, {
-      isComplete: isComplete,
-    })
-    location.reload()
-  } catch (error) {
-    console.error(error)
-  }
-}
 
-const deleteTodo = async (id: string) => {
-  try {
-    await axios.delete<TodoData>(
-      `https://calm-plum-jaguar-tutu.cyclic.app/todos/${id}`
-    )
-    location.reload()
-  } catch (error) {
-    console.error(error)
+const sortTodo = (sortType: string) => {
+  if (sortType === "asc") {
+    allData.value.sort((a, b) => a.todoName.localeCompare(b.todoName));
+  } else {
+    allData.value.sort((a, b) => b.todoName.localeCompare(a.todoName));
   }
-}
+};
```

Comment:
This might be better as a computed that doesn't mutate the original array, that way you can get back to the original default unsorted order if you want. It can also make code easier to follow if you generally go with computed over mutating a ref that way you can always be sure what the contents of your ref are without fear of something having been changed.